NeoSwift/NeoClient.swift

```diff
@@ -52,181 +77,269 @@ public class NeoClient {
+                completion(result)
+                break
             }
-            let data = try! JSONSerialization.data(withJSONObject: (json!["result"] as! [String: Any]), options: .prettyPrinted)
-            let transaction = try! decoder.decode(Transaction.self, from: data)
-            completion(transaction, error)
         }
     }
     
     //NEED TO GUARD ON THE VALUE OUTS
-    public func getTransactionOutput(with hash: String, and index: Int64, completion: @escaping (ValueOut?, Error?) -> Void) {
-        sendRequest(.getTransaction, params: [hash, index]) { json, error in
-            let decoder = JSONDecoder()
-            guard error == nil else {
-                completion(nil, error)
-                return
+    public func getTransactionOutput(with hash: String, and index: Int64, completion: @escaping (NeoClientResult<ValueOut>) -> ()) {
+        sendRequest(.getTransaction, params: [hash, index]) { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+                break
+            case .success(let response):
+                let decoder = JSONDecoder()
+                guard let data = try? JSONSerialization.data(withJSONObject: (response["result"] as! JSONDictionary), options: .prettyPrinted),
+                    let block = try? decoder.decode(ValueOut.self, from: data) else {
+                        completion(.failure(NeoClientError.invalidData))
+                        return
+                }
+                
+                let result = NeoClientResult.success(block)
+                completion(result)
+                break
             }
-            let data = try! JSONSerialization.data(withJSONObject: (json!["result"]), options: .prettyPrinted)
-            let transaction = try! decoder.decode(ValueOut.self, from: data)
-            completion(transaction, error)
         }
     }
     
-    public func getUnconfirmedTransactions(completion: @escaping ([String]?, Error?) -> Void) {
-        sendRequest(.getUnconfirmedTransactions, params: nil) { json, error in
-            guard error == nil,
-                let count = json?["result"] as? [String] else {
-                    completion(nil, error)
+    public func getUnconfirmedTransactions(completion: @escaping (NeoClientResult<[String]>) -> ()) {
+        sendRequest(.getUnconfirmedTransactions, params: nil) { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+                break
+            case .success(let response):
+                guard let txs = response["result"] as? [String] else {
+                    completion(.failure(NeoClientError.invalidData))
                     return
+                }
+                
+                let result = NeoClientResult.success(txs)
+                completion(result)
+                
+                break
```

Comment:
why are we breaking here?

Reply:
We can remove it only because it's the last case of the switch ;)